fhir-term-graph/src/main/java/com/ibm/fhir/term/graph/loader/impl/SnoMedICD10MapTermGraphLoader.java

```diff
@@ -0,0 +1,207 @@
+/*
+ * (C) Copyright IBM Corp. 2021, 2021
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package com.ibm.fhir.term.graph.loader.impl;
+
+import static com.ibm.fhir.term.graph.loader.util.FHIRTermGraphLoaderUtil.toMap;
+
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import java.util.concurrent.atomic.AtomicInteger;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+
+import org.apache.commons.cli.CommandLine;
+import org.apache.commons.cli.CommandLineParser;
+import org.apache.commons.cli.DefaultParser;
+import org.apache.commons.cli.HelpFormatter;
+import org.apache.commons.cli.MissingOptionException;
+import org.apache.commons.cli.Options;
+import org.apache.commons.configuration.Configuration;
+import org.apache.tinkerpop.gremlin.structure.Vertex;
+import org.janusgraph.core.schema.JanusGraphManagement;
+
+import com.ibm.fhir.term.graph.loader.FHIRTermGraphLoader;
+import com.ibm.fhir.term.graph.loader.util.COSObject;
+import com.ibm.fhir.term.graph.loader.util.ConfigLoader;
+
+/*
+ * This class will load edges between Snomed codes and ICD10 codes based on the UMLS into a JanusGraph.
+ */
+public class SnoMedICD10MapTermGraphLoader extends AbstractTermGraphLoader {
+    private static final Logger LOG = Logger.getLogger(SnoMedICD10MapTermGraphLoader.class.getName());
+
+    public static final String MAPS_TO = "mapsTo";
+
+    private static final String SNOMED_TO_ICD_MAP_FILE = "der2_iisssccRefset_ExtendedMapFull_US1000124_20210901.txt";
+
+    private static final String UMLS_DELIMITER = "\t";
+
+    /**
+     * Load the Snomed->ICD map from file, respecting effectiveDate and active status
+     * 
+     * @param cosBucketName
+     * @return a map from Snomed code to ICD10 code(s)
+     * @throws IOException
+     */
+    public static final Map<String,Set<String>> loadMap(String cosBucketName) throws IOException {
+        // For a given snomed code, find the most recent active row. If the rule is
+        // always map to a single ICD code, add an edge. If not, skip that snomed code.
+
+        Map<String, String> snomedToDateMap = new HashMap<>(); // Snomed->ICD, date
+        Map<String, Set<String>> snomedToICDMap = new HashMap<>(); // Snomed->ICD, date
+
+        final AtomicInteger rowCount = new AtomicInteger(0);
+        try (BufferedReader reader = new BufferedReader(COSObject.getItem(cosBucketName, SNOMED_TO_ICD_MAP_FILE))) {
```

Comment:
you mention that it should be closed by the Client. 
it should be InputStreamReader ir = getItem(); then the buffered reader

Reply:
The implied close of BufferedReader in the try-with-resources will close the underlying InputStreamReader.